<launch>

  <!-- Load temoto parameters -->
  <rosparam command="load" file="$(find temoto)/config/ripley.yaml" />


  <!-- Most of the action happens here -->
  <node name="start_teleop" pkg="temoto" type="start_teleop" launch-prefix="gnome-terminal --command" />


  <!-- Bring up speech-to-text and text-to-speech -->
  <!-- Several options here: PocketSphinx, Google speech-to-text, keyboard, ... -->
  <!-- Optional if you use buttons on the controller instead -->
  <!-- <include file="$(find pocketsphinx)/launch/pocketsphinx.launch" /> -->
  <!-- <include file="$(find google_speech_to_text)/launch/google_speech_to_text.launch" /> -->
  <node name="kb_fake_voice_commander" pkg="temoto" type="kb_fake_voice_commander" output="screen" />
  <node name="keyboard_event_publisher" pkg="keyboard_reader" type="keyboard_event_publisher"/>


  <!-- Pose controller -->
  <node name="xbox_joy" pkg="joy" type="joy_node" >
    <param name="autorepeat_rate" type="double" value="100" />
  </node>
  <include file="$(find spacenav_node)/launch/classic.launch"/>


  <!-- Bring up Griffin Powermate (OPTIONAL - for motion scaling) -->
  <!-- <node name="griffin_powermate" pkg="griffin_powermate" type="griffin_powermate" /> -->
  

  <!-- bringup proper rviz config -->
  <include file="$(find temoto)/launch/vaultbot_moveit_rviz.launch">
    <arg name="config" value="true"/>
  </include>


  <!-- Open an rqt window for camera feeds -->
  <node name="rqt_gui" pkg="vaultbot_drivers" type="rqt_gui.sh" />


  <!-- Start soundplay_node that is responsible for voice synthesis -->
  <node name="soundplay_node" pkg="sound_play" type="soundplay_node.py" />


  <!-- Arm jogging -->
  <include file="$(find vaultbot_drivers)/launch/jog_arms.launch" />


  <!-- Launch the EE levels/Angles to be displayed in RQT_MANIP of rqt_gui -->
  <!-- include file="$(find rqt_manip)/launch/ee_levels.launch" / -->


  <!-- Display joint angle warnings (OPTIONAL) -->
  <node name="arm_warning_left" pkg="arm_warnings" type="arm_warnings" output="screen" respawn="true">
    <param name="move_group_name" type="str" value="left_ur5" />
    <param name="marker_id" type="int" value="573" />

    <param name="minimum_joint_rad" type="double" value="-3.141" />
    <param name="maximum_joint_rad" type="double" value="3.141" />
  </node>

  <node name="arm_warning_right" pkg="arm_warnings" type="arm_warnings" output="screen" respawn="true">
    <param name="move_group_name" type="str" value="right_ur5" />
    <param name="marker_id" type="int" value="574" />

    <param name="minimum_joint_rad" type="double" value="-3.141" />
    <param name="maximum_joint_rad" type="double" value="3.141" />
  </node>

</launch>
